### PR TITLE
Fix generated Pulumi.yaml on convert

### LINKED ISSUE #10795: convert - `configuration`, `resources`, `outputs` blocks are being written to generated `Pulumi.yaml` files
### What happened?

`pulumi convert....` is writing `configuration`, `resources`, `outputs` blocks, from the `pulumi-yaml` source Pulumi.yaml file, to the outputted Pulumi.yaml files.

### Steps to reproduce

Execute below yaml in a Pulumi yaml project
```yaml
name: yaml-config-convert
runtime: yaml
description: A minimal Pulumi YAML program
configuration:
  val: 
    type: String
resources:
  bucket:
    type: aws:s3:Bucket
outputs:
  bucketName: ${bucket.bucket}
```

### Expected Behavior

Generated Pulumi.yaml files only contain relevant information to the generated code/project.

### Actual Behavior

Generated Pulumi.yaml files contain yaml blocks from source Pulumi.yaml file.

```yaml
name: yaml-config-convert
runtime: go
description: A minimal Pulumi YAML program
configuration:
  val:
    type: String
outputs:
  bucketName: ${bucket.bucket}
resources:
  bucket:
    type: aws:s3:Bucket
```

### Output of `pulumi about`

_No response_

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by aq17**
# Description

Fixes #10795 

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works

- [ ] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/codegen/dotnet/gen_program.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 )
 
+var ProjectKeysToOmit = []string{"configuration", "resources", "outputs"}
```
Comment (Frassle) [inline pkg/codegen/dotnet/gen_program.go:37]: I think we should be omitting everything except the keys pulumi explicitly knows about.

File: pkg/codegen/go/gen_program.go [hunk 2]
```diff
@@ -149,6 +151,11 @@ func GenerateProject(directory string, project workspace.Project, program *pcl.P
 
 	// Set the runtime to "go" then marshal to Pulumi.yaml
 	project.Runtime = workspace.NewProjectRuntimeInfo("go", nil)
+	if project.AdditionalKeys != nil {
```
Comment (Frassle) [inline pkg/codegen/go/gen_program.go:154]: Bit lame having to do this project clean in each languages codegen, can't the convert code clean the project before passing it down to codegen?

File: pkg/codegen/dotnet/gen_program.go (hunk 1 above)
Comment (iwahbe) [reply pkg/codegen/dotnet/gen_program.go:37]: I think we should respect `project.AdditionalKeys`. I would put this change in `"github.com/pulumi/pulumi-yaml/pkg/pulumiyaml/codegen".Eject`.

Comment (Frassle) [reply pkg/codegen/dotnet/gen_program.go:37]: hmm yes that's probably a better call actually, I didn't realize eject had a chance to change the project.

**Comment by aq17** [comment]
Closing in favor of https://github.com/pulumi/pulumi-yaml/pull/339.

**Pull request closed without merging**